### ISSUE TITLE
Add a section on how to debug proofs that check for memory leak

### ISSUE THREAD

**Issue by feliperodri** [model-checking/cbmc-training#9]
Currently, there is nothing in our training material about memory leaks.

**Comment by tautschnig** [comment]
Perhaps rightly so: these are not undefined behaviour.

**Comment by feliperodri** [comment]
We have proofs that make use of `--memory-leak-check`. For instance, take a look at [s2n_free_object](https://github.com/aws/s2n-tls/blob/8556f56bc6386dc8bfd8cceb6a38b97215a6f32e/tests/cbmc/proofs/s2n_free_object/Makefile#L14). If developers could use this option, we should document the particularities on how to debug these proofs or when it is useful to use this flag, right?

**Comment by tautschnig** [comment]
I agree that we need to provide better support for this, but perhaps it's something that we need to fix on the CBMC side (where https://github.com/diffblue/cbmc/issues/6073 exists, as you will be aware)?
